### DISCUSSION TITLE
TLS connection to db server?

### DISCUSSION THREAD

**Discussion by pettersillander** [sqlpage/SQLPage#299] [General]
Could implemeting (maybe optional?) encrypted connection between client and server help to ease adoption of SQLpage in diverse environments?

Quite a few networks (even closed) are encrypted connections only nowadays. As all the SQLpage supported dbms's support TLS natively it's seems rather cumbersome to use a tunnel for encryption.

Just wondering this after configuring SQLpage in existing environments...

**Comment by lovasoa** [comment]
Hello @pettersillander and welcome to SQLPage !

  I have great news for you... TLS is already supported in SQLPage today :slightly_smiling_face:  You can set advanced encryption options in the database connection string if you need. For instance, if you have a postgres database that supports ssl, SQLPage will connect to it over ssl by default. If you need SQLPage to refuse connecting over plain text connections, you can use `postgres://user:password@host/db?sslmode=require` as the database connection string.

Did you have an issue with ssl connection with a database in particular ? If so, can you share reproduction steps ?

**Reply by pettersillander** [reply]
@lovasoa I was trying to connect to 

Server: PostgreSQL 15.6 with clientcert=verify-ca in pg_hba.conf 
SQLPage: postgres://.../...?sslmode=require in sqlpage.json. 

Right now I can't say what actually failed, handshake or connection part.

When connection failed I was looking at [configuration.md](https://github.com/lovasoa/SQLpage/blob/main/configuration.md). I couldn't find any mentions of ssl use case there. So I basically didn't investigate further and went on with the tunneling.

But if TLS should work I'll try it again next time I'm at the site. I'll post an update then.

**Reply by lovasoa** [reply]
SSL connexions and server certificate validation are supported, but client certificates are not. 

Would you be interested in backporting https://github.com/launchbadge/sqlx/pull/1850 into https://github.com/lovasoa/sqlx-oldapi ?

**Reply by lovasoa** [reply]
Ok, I'm working on it: https://github.com/lovasoa/sqlx-oldapi/pull/7

**Reply by pettersillander** [reply]
@lovasoa This is great! I see you already included this in latest release. And it seems that client certificates are now working as well. 👍
